### DISCUSSION TITLE
Errors thrown in `nextHandler` passed to subscribe should not tear down observable

### DISCUSSION THREAD

**Discussion by benlesh** [ReactiveX/rxjs#6327] [General]
This is coming from the TC39 proposal (cc @jhusain)

Basically, to match the behavior of EventTarget, an error in the handler shouldn't cause the source to stop emitting values. In other words:

```js
Observable.interval(1000)
  .subscribe({
    next(x) {
      if (x === 1) throw new Error('bad');
      console.log(x);
    }
  });

// 0
// ERROR: bad [callstack vomit here]
// 1
// 2
// 3
// ...
```

**Comment by mattpodwysocki** [comment]
@blesh as far as I remember, it stops emitting, but doesn't tear down the Observable, so not quite what you have there.  @jhusain am I correct here?

**Comment by jhusain** [comment]
Keeps emitting, doesn't tear down Observable. The forEach method however
continues to work the way subscribe works today.

On Thu, Dec 1, 2016 at 7:58 PM, Matthew Podwysocki <notifications@github.com

**Comment by benlesh** [comment]
I think we're going to have to target this for v7. It's an important change, but it will incur a lot of breakage in the community I think.
